README.md

```diff
@@ -62,6 +65,66 @@ dsc_windowsfeature {'IIS':
 
 All DSC Resource names and parameters have to be in lowercase, e.g: `dsc_windowsfeature` or `dsc_name`.
 
+### Installing Packages with DSC
+
+You can install MSIs with DSC using the Puppet type `dsc_package` which maps to the `Package` DSC Resource.
+
+~~~puppet
+dsc_package{'installpython'
+  dsc_ensure    => 'Present',
+  dsc_name      => 'Python 2.7.10',
+  dsc_productid => 'E2B51919-207A-43EB-AE78-733F9C6797C2'
+  dsc_path      => 'C:\\python.msi',
+}
+~~~
+
+The `Package` DSC Resource requires the following pieces of information to install an MSI:
+
+- ProductName: The `Name` of product being installed.
+- ProductId: The `ProductCode` of the MSI, which is the unique identifier for that MSI version.
+
+You can obtain this information in a variety of ways.
+
+- Use a tool like Orca to open the MSI file and inspect the `Name` and `ProductCode`.
+- Install the product on a test system and inspect the `Name` and `ProductCode` in the Windows Add/Remove Programs Control Panel
+- Use a script to query the MSI file for the `Name` and `ProductCode`.
+
+~~~powershell
+function Get-MsiDatabaseVersion
+{
+  param (
+    [IO.FileInfo]$FilePath
+  )
+
+  $productName = Invoke-MSIQuery -query "SELECT Value FROM Property WHERE Property = 'ProductName'"
+  $productCode = Invoke-MSIQuery -query "SELECT Value FROM Property WHERE Property = 'ProductCode'"
+
+  return [PSCustomObject]@{
+    FullName       = $FilePath
+    ProductName    = ([string]$productName).TrimStart()
+    ProductCode    = ([string]$productCode).Replace("{","").Replace("}","").TrimStart()
+  }
+}
+
+function Invoke-MSIQuery
+{
+  param($query)
+  try {
+    $windowsInstaller = New-Object -com WindowsInstaller.Installer
+    $database = $windowsInstaller.GetType().InvokeMember("OpenDatabase", "InvokeMethod", $Null, $windowsInstaller, @($FilePath.FullName, 0))
+    $View = $database.GetType().InvokeMember("OpenView", "InvokeMethod", $Null, $database, ($query))
+    $View.GetType().InvokeMember("Execute", "InvokeMethod", $Null, $View, $Null)
+    $record = $View.GetType().InvokeMember("Fetch", "InvokeMethod", $Null, $View, $Null)
+    $propertyValue = $record.GetType().InvokeMember("StringData", "GetProperty", $Null, $record, 1)
+    return $propertyValue
+  } catch {
+    throw "Failed to get MSI file version the error was: {0}." -f $_
+  }finally{
+    $database.Dispose()
```

Comment:
nit, $database can be null if either line 113 or 114 raise an exception. Best to do something like:

```
$windowsInstaller = New-Object -com WindowsInstaller.Installer
$database = $windowsInstaller.GetType().InvokeMember("OpenDatabase", "InvokeMethod", $Null, $windowsInstaller, @($FilePath.FullName, 0))
try {
  ..
} finally {
  $database.Dispose()
}
```